Makefile

```diff
@@ -329,16 +332,9 @@ endif
 endif
 
 package-minimal-headless-wasmer:
-ifdef cross_compiling_to_mac_aarch64
-	if [ -f "target/aarch64-apple-darwin/release/wasmer" ]; then \
-		cp target/aarch64-apple-darwin/release/wasmer package/bin/wasmer-headless ;\
+	if [ -f "target/$(HOST_TARGET)/release/wasmer-headless" ]; then \
```

Comment:
Don't we still need the special casing for `aarch64` macos here? if we're not building on actual aarch64 macos machines, I believe this will not work correctly

Reply:
Not sure if I follow why.
`HOST_TARGET` will be `aarch64-apple-darwin` when cross compiling to `macos-arm64` since we setup the target in cargo config:
https://github.com/wasmerio/wasmer/blob/master/.github/workflows/main.yaml#L138-L144

Am I missing something?

Reply:
I would assume
```
 HOST_TARGET=$(shell rustup show | grep 'Default host: ' | cut -d':' -f2 | tr -d ' ') 
 ```
 
 Will not give the correct results there, but I'm just assuming

Reply:
I just re-checked and you are right, however I just found out the previous logic was also wrong.

```cmake
cross_compiling_to_mac_aarch64 =
ifneq (,$(filter $(ARCH),aarch64 arm64))
	test_compilers_engines += cranelift-jit
	ifneq (, $(findstring llvm,$(compilers)))
		test_compilers_engines += llvm-native
	endif
	# if we are in macos arm64, we use the system libffi for the capi
	ifeq ($(UNAME_S), Darwin)
		cross_compiling_to_mac_aarch64 = yes
		use_system_ffi = yes
	endif
endif
```

This can never be true. `cross_compiling_to_mac_aarch64` will never be yes unless the host is a macOS M1 itself

Reply:
Re-checking this: 

```
HOST_TARGET=$(shell rustup show | grep 'Default host: ' | cut -d':' -f2 | tr -d ' ') 
```

Should return `aarch64-apple-darwin` when cross-compiling. Since it's the default (and only) one installed for that.
So we should be good!